File: ps/can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int i;
        bool canFlowered;
        for(i=0;i<flowerbed.size();i++) {
            canFlowered = flowerbed[i] ? false : true;
            if (i != 0) canFlowered &= flowerbed[i-1] ? false : true;
            if (i != flowerbed.size()-1) canFlowered &= flowerbed[i+1] ? false : true;
            
            if (canFlowered) {
                flowerbed[i] = 1;
                n--;
            }
        }
        return n<=0?true:false;
    }
};

File: ps/can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        for i in range(len(flowerbed)):
            canFlowered = True
            if i != 0 and flowerbed[i - 1] == 1:
                canFlowered = False
            if flowerbed[i] == 1:
                canFlowered = False
            if i != len(flowerbed) - 1 and flowerbed[i + 1] == 1:
                canFlowered = False

            if canFlowered:
                flowerbed[i] = 1
                n -= 1

        if n <= 0:
            return True

        return False